src/components/form/examples/form-span-fields.tsx

```diff
@@ -0,0 +1,215 @@
+ * values higher than 5 (or higher than maximum number of columns in the form)
+ * will only force the file to be full-width, just like what `all` does.
+ *
+ * ````ts
+ * export const schema = {
+ *     …
+ *     properties: {
+ *         name: {
+ *             type: 'string',
+ *             title: 'Comment',
+ *             lime: {
+ *                 layout: {
+ *                       colSpan: 'all',
+ *                   },
+ *               },
+ *         },
+ *         …
+ *     },
+ *     …
+ * };
+ * ````
+ *
+ * ###### Dense layout (Auto reorder fields to avoid empty cells)
+ * Order of fields and number of columns that a field must span through, can
+ * consequently affect the layout of your responsive form when the screen width
+ * changes.
+ * Let's say you have a form with a 4 columned layout, and you specify that its
+ * second field must span (or stretch) through 3 columns.
+ * Now if the container's width decreases, it will forces the form to render its
+ * layout in 3 columns instead. Therefore, the second field has to jump
+ * to the next line to still be able to stretch through 3 columns.
+ * This will leave 2 empty cells in the first row, right after the first field.
+ *
+ * To avoid having these empty cells in the UI, limel-form will automatically
+ * bring up two other fields that span only 1 column, and places them in the
+ * available empty cells in the first row.
+ *
+ * However, you can disable this functionality by setting `dense` to fall in the
+ * options for the grid layout
+ *
+ * ```ts
+ * export const schema = {
+ *     type: 'object',
+ *     lime: {
+ *         layout: {
+ *             type: 'grid',
+ *             dense: false,
+ *         },
+ *     },
+ * };
+ * ```
+ *
+ * :::note
+ * Sometimes order of fields are important in the way users perceive the form.
+ * If you choose to have the default auto-reordering behavior of the forms,
+ * make sure to test yout forms layout in different screen sizes to see whether
+ * you can mitigate unwanted layout changes.
+ *
+ * You can avoid unwanted results by changing order of fields so that they render
+ * appropriately on different screens, or by dividing the form into more
+ * sections.
+ * :::
+ *
+ * ###### Stretching a field vertically
+ * All standard elements that can be used in forms such as `limel-input`,
+ * `limel-select`, `limel-slider`, etc. have a fixed UI in the horizontal axis
+ * and therefore it does not really make sense to stretch them in height,
+ * and we strongly recommend you not to!
+ *
+ * But there are some exceptions. One of them is the `textarea`.
+ * It's always recommended to assign at least two rows for a textarea, to improve
+ * its usability. Since this is a component that is supposed to show multi-lined
+ * pieces of text.
+ *
+ * Also, if you make create a custom component for your form –let's say a map–,
+ * you will have the possibility of using `rowSpan`, to increase the height of
+ * your custom component.
```

Comment:
```suggestion
 * Also, if you make create a custom component for your form—let's say
 * a map—you will have the possibility of using `rowSpan` to increase the height of
 * your custom component.
```

Reply:
It's not visible in the code, but those "hyphens" are this kind: —

I find it ugly myself, but the "correct" way of using them is actually _without_ spaces… and they shouldn't be used with commas, as they act as a kind of "super-comma".

Reply:
so there should be a `space` after them, not before them?

Reply:
What I have learnt is that: when you put something between these `–` characters, it means the meaning with or without "that thing which is between the `–`s" will have the same message/meaning. In other words, you can drop that part without making your sentence meaningless (it's like an inline footnote). But if your original sentence has commas, then commas should still be where they are with or without the "inline footnotes". But I have no academic reference to support wha I'm saying. So feel free to change ;)

Reply:
No spaces around them at all (which is why I had to put the line-break between the words inside them).

More info here: https://www.dictionary.com/e/em-dash/

Reading the _When to use an em dash_ section on that page, it seems like it would be more correct to just use commas in the instance above… 🤔

Reply:
> What I have learnt is that: when you put something between these `–` characters, it means the meaning with or without "that thing which is between the `–`s" will have the same message/meaning. In other words, you can drop that part without making your sentence meaningless (it's like an inline footnote). But if your original sentence has commas, then commas should still be where they are with or without the "inline footnotes". But I have no academic reference to support wha I'm saying. So feel free to change ;)

My only knowledge of how to use em dashes is from reading English language newspaper articles. Newspapers love using them, while seemingly no one else does 😄

And yes, I think I've mostly seen them used as you describe, infixing a kind of footnote. Which is why I was a little surprised to not see that use listed on dictionary.com. I say keep the em dashes! 👍